src/backend/marsha/bbb/api.py

```diff
@@ -636,7 +636,9 @@ def create_vod(self, request, pk=None, classroom_id=None):
             HttpResponse with the serialized classroom recording vod.
 
         """
-        classroom_recording = self.get_object()  # check permissions first
+        classroom_recording = ClassroomRecording.objects.select_related(
+            "classroom__playlist__consumer_site"
+        ).get(id=pk)
```

Comment:
Doing this permissions are not check anymore. An exception will be raised if the `pk` does not exists instead of returning a `404`.
IMO you have to override the `get_queryset` method and do this in it if the action is `create_vod`.